apps/src/templates/sectionProgress/progressTables/ProgressTableLevelBubble.jsx

```diff
@@ -3,181 +3,146 @@ import Radium from 'radium';
-  };
-}
-
-function mainStyle(props) {
-  return {
-    ...styles.main,
-    ...levelStyle(props)
-  };
-}
-
-function largeStyle(props) {
-  const {disabled, bonus} = props;
-  return {
-    ...mainStyle(props),
-    ...styles.large,
-    ...(disabled && bonus && styles.bonusDisabled)
+import {
+  bubbleStyles,
+  BubbleSize,
+  BubbleShape,
+  mainBubbleStyle,
+  diamondContainerStyle,
+  getProgressStyle
+} from '@cdo/apps/templates/progress/progressStyles';
+
+/**
+ * Top-level bubble component responsible for configuring BasicBubble based on
+ * properties of a level and student progress.
+ */
+class ProgressTableLevelBubble extends React.PureComponent {
+  static propTypes = {
+    levelStatus: PropTypes.string,
+    levelKind: PropTypes.string,
+    isDisabled: PropTypes.bool,
+    isUnplugged: PropTypes.bool,
+    isConcept: PropTypes.bool,
+    isBonus: PropTypes.bool,
+    isPaired: PropTypes.bool,
+    bubbleSize: PropTypes.oneOf(Object.values(BubbleSize)).isRequired,
+    title: PropTypes.string,
+    url: PropTypes.string
   };
-}
 
-function largeContentStyle(props) {
-  const {bonus, paired} = props;
-  return {
-    ...progressStyles.flex,
-    ...styles.contents,
-    fontSize: paired || bonus ? 14 : 16
+  static defaultProps = {
+    bubbleSize: BubbleSize.full
   };
-}
 
-function UnpluggedBubble(props) {
-  return (
-    <div
-      style={{
-        ...styles.main,
-        ...styles.unplugged,
-        ...levelStyle(props)
-      }}
-    >
-      <Content {...props} />
-    </div>
-  );
-}
+  renderContent() {
+    const {
+      levelStatus,
+      isUnplugged,
+      isBonus,
+      isPaired,
+      title,
+      bubbleSize
+    } = this.props;
+    if (bubbleSize === BubbleSize.dot) {
+      // dot-sized bubbles are too small for content
+      return null;
+    }
+    const isLocked = levelStatus === LevelStatus.locked;
+    return isUnplugged ? (
+      <span>{i18n.unpluggedActivity()}</span>
+    ) : isLocked ? (
+      <FontAwesome icon="lock" />
+    ) : isPaired ? (
+      <FontAwesome icon="users" />
+    ) : isBonus ? (
+      <FontAwesome icon="flag-checkered" />
+    ) : title ? (
+      <span>{title}</span>
+    ) : null;
+  }
 
-function SmallCircle(props) {
-  return (
-    <div style={{...mainStyle(props), ...styles.smallCircle}}>
-      <Content {...props} />
-    </div>
-  );
+  render() {
+    const {isUnplugged, isConcept, isDisabled, bubbleSize} = this.props;
+    const bubbleShape = isUnplugged
+      ? BubbleShape.pill
+      : isConcept
+      ? BubbleShape.diamond
+      : BubbleShape.circle;
+
+    const bubble = (
+      <BasicBubble
+        shape={bubbleShape}
+        size={bubbleSize}
+        progressStyle={getProgressStyle(this.props)}
+      >
+        {this.renderContent()}
+      </BasicBubble>
+    );
+
+    if (isDisabled) {
+      return bubble;
+    }
+    return <LinkWrapper {...this.props}>{bubble}</LinkWrapper>;
```

Comment:
(unrelated, for the future) It would be nice if we passed just the url property through to LinkWrapper (instead of all the properties).